pkg/storageincentives/agent.go

```diff
@@ -413,13 +464,60 @@ func (a *Agent) Close() error {
 	}
 }
 
-func (s *Agent) wrapCommit(storageRadius uint8, sample []byte, key []byte) ([]byte, error) {
+func (a *Agent) wrapCommit(storageRadius uint8, sample []byte, key []byte) ([]byte, error) {
 
 	storageRadiusByte := []byte{storageRadius}
 
-	data := append(s.overlay.Bytes(), storageRadiusByte...)
+	data := append(a.overlay.Bytes(), storageRadiusByte...)
 	data = append(data, sample...)
 	data = append(data, key...)
 
 	return crypto.LegacyKeccak256(data)
 }
+
+// setNodeStatus sets the internal node status
+func (a *Agent) setNodeStatusFee(fee *big.Int) {
+	a.mtx.Lock()
+	defer a.mtx.Unlock()
+	if fee != nil {
+		a.nodeStatus.Fees = a.nodeStatus.Fees.Add(a.nodeStatus.Fees, fee)
+	}
+}
+
+// calculateWinnerReward calculates the reward for the winner
+func (a *Agent) calculateWinnerReward() error {
+	currentBalance, err := a.erc20Service.BalanceOf(context.Background(), common.HexToAddress(a.overlay.String()))
+	if err != nil {
+		return err
+	}
+	if currentBalance != nil {
+		a.nodeStatus.Reward = currentBalance.Sub(currentBalance, a.initialBalance)
+	}
+	return nil
+}
+
+// saveStatus saves the node status to the database
+func (a *Agent) saveStatus() {
+	a.mtx.Lock()
+	defer a.mtx.Unlock()
+	err := a.nodeState.Put(fmt.Sprintf("%s%s", redistributionStatusKey, a.overlay.String()), a.nodeStatus)
+	if err != nil {
+		a.logger.Error(err, "error saving node status")
+		return
+	}
+}
+
+// GetStatus returns the node status
+func (a *Agent) GetStatus() (status NodeStatus, err error) {
+	return status, a.nodeState.Get(fmt.Sprintf("%s%s", redistributionStatusKey, a.overlay.String()), &status)
```

Comment:
you do not overlay as part of the key

Reply:
I am removing the overlay from this in new implementation.